src/main/java/umaikaze/duke/Duke.java

```diff
@@ -0,0 +1,74 @@
+package umaikaze.duke;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.time.format.DateTimeParseException;
+
+public class Duke {
+    Ui ui;
+    Storage st;
+    TaskList tl;
+
+    public Duke(String savePath) throws UnsupportedEncodingException {
+        ui = new Ui();
+        st = new Storage(savePath);
+        try {
+            tl = new TaskList(st.loadFile());
+        } catch (IOException e) {
+            ui.showError(e.getMessage());
+            tl = new TaskList();
+        }
+    }
+
+    void run(BufferedReader br) throws IOException {
+        ui.showReply("how may i sewve u today nya?");
+        String[] line = br.readLine().split(" ");
+        String cmd = line[0];
+        while (!cmd.equals("bye")) {
+            try {
+                switch (cmd) {
+                    case "list":
+                        tl.printList(ui);
+                        break;
+                    case "done":
+                        ui.showReply(tl.markDone(Integer.parseInt(line[1]) - 1));
+                        break;
+                    case "delete":
+                        ui.showReply(tl.delete(Integer.parseInt(line[1]) - 1));
+                        break;
+                    default:
+                        ui.showReply(tl.addTask(line));
+                        break;
+                }
+                if (!cmd.equals("list")) {
+                    tl.saveFile(st);
+                }
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            } catch (DateTimeParseException e) {
+                ui.showError(e.getMessage() + "\n\tYouw date and time fowmat is invawid ^;;w;;^ "
```

Comment:
Are words like `invawid`, `suwe`, and `unyabwe` typo, or they are meant to be cute? 😄 
You might consider recheck the spelling so your users won't get confused.